src/components/VideoProvider/useBackgroundSettings/useBackgroundSettings.ts

```diff
@@ -65,13 +68,61 @@ export const backgroundConfig = {
   images,
 };
 
-// TODO : Add video processing logic after backgroundSettings change
-// useEffect hooks, etc ...
+const virtualBackgroundAssets = '/virtualbackground';
 
-export default function useBackgroundSettings() {
-  const [backgroundSettings, setBackgroundSettings] = useState({
-    type: 'none',
-    index: 0,
-  } as BackgroundSettings);
-  return [backgroundSettings, setBackgroundSettings] as const;
+let blurProcessor: GaussianBlurBackgroundProcessor;
+
+export default function useBackgroundSettings(videoTrack: LocalVideoTrack) {
+  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(() => {
+    const localStorageSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
+    return localStorageSettings ? JSON.parse(localStorageSettings) : { type: 'none', index: 0 };
+  });
+
+  // load processors on initialization
+  useEffect(() => {
+    if (!blurProcessor) {
+      blurProcessor = new GaussianBlurBackgroundProcessor({
+        assetsPath: virtualBackgroundAssets,
+      });
+      blurProcessor.loadModel();
```

Comment:
I'm not sure that this needs to be in a useEffect hook. Would it work if we moved it outside of `useBackgroundSettings`?